### PR TITLE
Traditional Grid Pagination

### LINKED ISSUE #934: Grid Pagination
**Enhancement**

Add traditional pagination to the grid as an opt-in feature that can be configured instead of the default infinite scrolling. The pagination should be able to be controlled via controls in the grid footer.

Adding an additional optional `pagination` property to the `Grid` API that will be used to turn on the feature.

```tsx
<Grid 
    fetcher={fetcher}
    pagination={true}
    columnConfig={columnConfig}
    height={450}
/>;
```

### PR DESCRIPTION
**Pull request by agubler**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/framework/blob/master/STYLE.md)
* [x] Unit tests are included in the PR
* [x] For new widgets, an entry has been added to the `.dojorc`
* [x] Any widget variant uses `theme.compose` like [this](https://github.com/dojo/widgets/issues/847)
* [x] WidgetProperties are exported

**Description:**

Add support for traditional grid pagination as an opt-in feature. Includes basic details and page navigation controls in the grid footer.

Resolves #934

### CODE REVIEW

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-muwmu5nkb.now.sh
* Tests Deployment: https://dojowidgets-qcx81229m.now.sh

File: src/grid/PaginatedFooter.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+	private _renderPaginationControls(totalPages: number) {
+		const { page } = this.properties;
+		if (page < 4) {
+			return [
+				this._renderPageControl(1),
+				this._renderPageControl(2),
+				this._renderPageControl(3),
+				this._renderPageControl(4),
+				this._renderPageControl(5),
+				v('span', { key: 'more', classes: [this.theme(css.more), fixedCss.moreFixed] }, [
```
Comment (tomdye) [inline src/grid/PaginatedFooter.tsx]: Could you add `aria-hidden` to these please?

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-bngdno79k.now.sh
* Tests Deployment: https://dojowidgets-3g1req55p.now.sh

**Comment by dojo-site** [comment]
This pull request has been deployed to:
* Docs Deployment: https://dojowidgets-5nfdk4jr3.now.sh
* Tests Deployment: https://dojowidgets-97n49sp9j.now.sh

**Pull request merged**